packages/warriorjs-core/src/Unit.js

```diff
@@ -157,7 +159,7 @@ class Unit {
   damage(receiver, amount) {
     receiver.takeDamage(amount);
     if (!receiver.isAlive()) {
-      this.earnPoints(receiver.maxHealth);
+      this.earnPoints(receiver.reward || receiver.maxHealth);
```

Comment:
@RascalTwo I prefer we make this check in the constructor, and here just award `receiver.reward`. Also, there could be scenarios where we want the reward to be `0`, which evaluates to false, so we may need to make this check more explicit (like `reward === null ? maxHealth : reward`).